src/sim/bridge.cc

```diff
@@ -0,0 +1,212 @@
+        LOG << "[SW-AXI] The bridge is already connected to " << connectedUri << std::endl;
+        return -1;
+    }
+
+    sockaddr_un addr;
+    if (path.length() > sizeof(addr.sun_path) - 1) {
+        LOG << "[SW-AXI] Path too long: " << path << std::endl;
+        return -1;
+    }
+
+    if (remove(path.c_str()) == -1 && errno != ENOENT) {
+        LOG << "[SW-AXI] Unable to remove: " << path << std::endl;
+        return -1;
+    }
+
+    sock = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (sock == -1) {
+        LOG << "[SW-AXI] Unable to create a UNIX socket: " << strerror(errno) << std::endl;
+        return -1;
+    }
+
+    memset(&addr, 0, sizeof(sockaddr_un));
+    addr.sun_family = AF_UNIX;
+    strncpy(addr.sun_path, path.c_str(), sizeof(addr.sun_path) - 1);
+
+    if (bind(sock, reinterpret_cast<sockaddr *>(&addr), sizeof(sockaddr_un)) == -1) {
+        LOG << "[SW-AXI] Unable to bind the socket: " << strerror(errno) << std::endl;
+        disconnect();
+        return -1;
+    }
+
+    if (listen(sock, 1) == -1) {
+        LOG << "[SW-AXI] Unable to listen for incomming connections: " << strerror(errno) << std::endl;
+        disconnect();
+        return -1;
+    }
+
+    int connSock = accept(sock, nullptr, nullptr);
+    if (connSock == -1) {
+        LOG << "[SW-AXI] Unable to accept a connection: " << strerror(errno) << std::endl;
+        disconnect();
+        return -1;
+    }
+
+    LOG << "[SW-AXI] Client connected to: " << uri << std::endl;
+    sock = connSock;
+    connectedUri = uri;
+
+    return 0;
+}
+
+int Bridge::sendSimulatorInfo() {
+    char hn[256];
+    hn[0] = 0;
+    gethostname(hn, 256);
+
+    flatbuffers::FlatBufferBuilder builder(1024);
+    auto name = builder.CreateString(std::string(STR1(SIM_ID)));
+    auto uri = builder.CreateString(connectedUri);
+    auto hostname = builder.CreateString(hn);
+    sw_axi::wire::SimulatorInfoBuilder siBuilder(builder);
+    siBuilder.add_simulatorName(name);
+    siBuilder.add_pid(getpid());
+    siBuilder.add_uri(uri);
+    siBuilder.add_hostname(hostname);
+    auto si = siBuilder.Finish();
+    builder.Finish(si);
+
+    if (sw_axi::writeToSocket(sock, builder.GetBufferPointer(), builder.GetSize()) == -1) {
+        return -1;
+    }
+    return 0;
+}
+
+void Bridge::disconnect() {
+    if (sock != -1) {
+        close(sock);
+        sock = -1;
+    }
+}
+
+void Bridge::waitForDisconnect() {
+    if (sock == -1) {
+        LOG << "[SW-AXI] Client not connected" << std::endl;
+        return;
+    }
+    LOG << "[SW-AXI] Waiting for the client to disconnect from: " << connectedUri << std::endl;
+
+    // TODO(lj): This is enough to track disconnection, to be replaced by a condition variable later
+    char buff;
+    read(sock, &buff, 1);
+    disconnect();
+
+    LOG << "[SW-AXI] Client disconnected from: " << connectedUri << std::endl;
+}
+}  // namespace
+
+extern "C" void *sw_axi_connect(const char *uri) {
+    Bridge *bridge = new Bridge;
```

Comment:
This is never freed, obviously. But maybe okay for the current use-case. Or this could be freed in sw_axi_disconnect

Reply:
Yup. Will fix. Thanks!